cpp/tests/test_abm_world.cpp

```diff
@@ -299,52 +299,47 @@ TEST(TestWorld, evolveMigration)
 
     // Test that a dead person cannot make a movement
     {
-        auto t      = mio::abm::TimePoint(0) + mio::abm::hours(8);
-        auto dt     = mio::abm::hours(11);
+        auto t      = mio::abm::TimePoint(0) + mio::abm::hours(3);
+        auto dt     = mio::abm::hours(1);
         auto params = mio::abm::GlobalInfectionParameters{};
-
-        auto world       = mio::abm::World(params);
-        auto home_id     = world.add_location(mio::abm::LocationType::Home);
-        auto work_id     = world.add_location(mio::abm::LocationType::Work);
-        auto hospital_id = world.add_location(mio::abm::LocationType::Hospital);
-        auto icu_id      = world.add_location(mio::abm::LocationType::ICU);
-
-        auto& p_dead = add_test_person(world, icu_id, mio::abm::AgeGroup::Age60to79, mio::abm::InfectionState::Dead, t);
-        auto& p_critical = add_test_person(world, hospital_id, mio::abm::AgeGroup::Age60to79,
-                                           mio::abm::InfectionState::InfectedCritical, t);
+        params.get<mio::abm::SevereToCritical>()[{mio::abm::VirusVariant::Wildtype, mio::abm::AgeGroup::Age60to79,
+                                                  mio::abm::VaccinationState::Unvaccinated}] = 0.01;
+        params.get<mio::abm::CriticalToDead>()[{mio::abm::VirusVariant::Wildtype, mio::abm::AgeGroup::Age60to79,
+                                                mio::abm::VaccinationState::Unvaccinated}]   = 0.01;
+
+        auto world   = mio::abm::World(params);
+        auto home_id = world.add_location(mio::abm::LocationType::Home);
+        auto work_id = world.add_location(mio::abm::LocationType::Work);
+        auto icu_id  = world.add_location(mio::abm::LocationType::ICU);
+        // Create a person that is dead at time t - dt
+        auto& p_dead =
+            add_test_person(world, icu_id, mio::abm::AgeGroup::Age60to79, mio::abm::InfectionState::Dead, t - dt);
+        // Create a person that will be dead at time t
+        auto& p_severe =
+            add_test_person(world, home_id, mio::abm::AgeGroup::Age60to79, mio::abm::InfectionState::Dead, t);
         p_dead.set_assigned_location(icu_id);
-        p_dead.set_assigned_location(home_id);
-        p_dead.set_assigned_location(work_id);
-        p_critical.set_assigned_location(hospital_id);
-        p_critical.set_assigned_location(icu_id);
-
-        p_critical.add_new_infection(mio::abm::Infection(mio::abm::VirusVariant::Wildtype,
-                                                         mio::abm::AgeGroup::Age60to79, params, t + dt,
-                                                         mio::abm::InfectionState::Dead));
+        p_severe.set_assigned_location(home_id);
```

Comment:
p_severe should also have a hospital and icu assigned shouldn't it?

Reply:
Yes, I corrected it accordingly.